pkg/storage/odf.go

```diff
@@ -0,0 +1,279 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+
+	odfoperatorv1alpha1 "github.com/red-hat-storage/odf-operator/api/v1alpha1"
+
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StorageSystemBuilder provides struct for storageSystem object containing connection
+// to the cluster and the storageSystem definitions.
+type StorageSystemBuilder struct {
```

Comment:
package is name storage. Let remove storage from the object name in order to remove duplication.
I think we also need to add odf in order to point to specific storage class.
Please re-name it to:
OdfSystemBuilder

Reply:
and both of the odf related

Reply:
Please refer to https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/storage/storageclass.go . The same logic we removed storage from object name in order to avoid duplication. Can we do the same wit this odf object?

Reply:
this is a point, I tried to explain to you when I worked on storageclass handler as well.
never mind, as you wish ...